nautobot/dcim/models/device_components.py

```diff
@@ -495,6 +499,22 @@ class BaseInterface(RelationshipModel):
         verbose_name="MTU",
     )
     mode = models.CharField(max_length=50, choices=InterfaceModeChoices, blank=True)
+    parent = models.ForeignKey(
+        to="self",
+        on_delete=models.SET_NULL,
+        related_name="child_interfaces",
+        null=True,
+        blank=True,
+        verbose_name="Parent interface",
+    )
```

Comment:
I think we should name this `parent_interface`. In many places we have `parent` accessing things like the Device or Virtual Machine, etc that these interfaces are associated with and this will incur a breaking change for many plugins/integrations.